File: windows/dart/lib/notify.dart
import 'dart:io' show Directory, Platform, Process, ProcessStartMode;
import 'package:path/path.dart' as path;
import 'consts.dart' show ssdir;

final osdigits = Platform.environment['ARCHITECTURE'];
/**
Usage: notifu [@argfile] [/?|h|help] [/v|version] [/t <value>] [/d ] [/p ]... [/m ]... [/i ] [/e] [/q] [/w] [/xp] [/c] [/k] [/l]

@argfile	Read arguments from a file.
/?		Show usage.
/v		Show version.
/t <value>	The type of message to display values are:
		   info         The message is an informational message
		   warn         The message is an warning message
		   error        The message is an error message
/d <value>	The number of milliseconds to display (omit or 0 for infinit)
/p <value>	The title (or prompt) of the ballon
/m <value>	The message text
/i <value>	Specify an icon to use ('parent' uses the icon of the parent process)
/e		Enable ballon tips in the registry (for this user only)
/q		Do not play a sound when the tooltip is displayed
/w		Show the tooltip even if the user is in the quiet period that follows his very first login (Windows 7 and up)
/xp		Force WindowsXP ballon tips behavior
/c		Constant (persistent) instance, will not be dismissed by new instances
/k		Removes every instance of Notifu, even the ones who are waiting to be displayed
/l		Diplay the license (BSD-3-Clause)
 */
void notify(String title, String message, [int timeout = 2500, String type = 'info']) {
  var cmd = path.join(ssdir, 'notifu${osdigits == '64' ? '64' : ''}.exe');
  Process.start(cmd, ['/t', type, '/d', timeout.toString(), '/p', title, '/m', message, '/q'], mode: ProcessStartMode.detached);
}

void main(List<String> args) {
  print(ssdir);
}

File: windows/node/notify.js
const os = require('os')
const path = require('path')
const WindowsBalloon = require('node-notifier').WindowsBalloon;
const { ssdir } = require('./ssbinconfig')

module.exports = function notify(message, exit = true) {
  notify.notifier = notify.notifier || new WindowsBalloon({
    withFallback: false, // Try Windows Toast and Growl first?
    customPath: path.join(ssdir, `notifu${os.arch() === 'x64' ? '64' : ''}.exe`) // Relative/Absolute path if you want to use your fork of notifu
  });
  notify.notifier.notify(
    {
      title: '✈️ SS Accounts',
      message: message || title,
      sound: false, // true | false.
      time: 1000, // How long to show balloon in ms
      wait: false, // Wait for User Action against Notification
      type: 'info' // The notification type : info | warn | error
    },
    function () {
      exit && setTimeout(() => {
        process.exit(0)
      }, 100);
    }
  );
}
